.github/workflows/ci-tk.yaml

```diff
@@ -32,12 +115,47 @@ jobs:
         os: [ubuntu-22.04, linux-mi300-1gpu-ossci-iree-org, nodai-amdgpu-mi250-x86-64]
     runs-on: ${{matrix.os}}
     timeout-minutes: 60
+    needs: build_llvm_linux
     if: github.event.pull_request.draft == false
     env:
       VENV_DIR: ${{ github.workspace }}/.turbine-venv
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
 
+      - name: Checkout Water
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        if: "(!contains(toJSON(matrix.os), 'amdgpu') && contains(toJSON(matrix.os), 'mi300')) && !cancelled()"
+        with:
+          repository: ftynse/water
+          ref: ${{ env.WATER_REF }}
+          path: water
+          fetch-depth: 0
+
+      - name: Setup Cache Vars
+        if: "(!contains(toJSON(matrix.os), 'amdgpu') && contains(toJSON(matrix.os), 'mi300')) && !cancelled()"
+        run: |
+          echo 'LLVM_SHA<<EOF' >> $GITHUB_ENV
+          cat ${GITHUB_WORKSPACE}/water/$LLVM_SHA_FILE >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+
+          echo 'WAVE_TEST_WATER<<EOF' >> $GITHUB_ENV
+          echo '1' >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
```

Comment:
What are these EOF values doing here? Could use a comment, or move this into a script.